services/src/pro/datasets/storage.rs

```diff
@@ -0,0 +1,197 @@
+use crate::datasets::storage::MetaDataDefinition;
+use crate::datasets::upload::UploadId;
+use crate::datasets::{AddDataset, DatasetIdAndName};
+use crate::error::Result;
+use async_trait::async_trait;
+use geoengine_datatypes::dataset::DatasetId;
+use geoengine_datatypes::primitives::DateTime;
+use postgres_types::{FromSql, ToSql};
+use serde::{Deserialize, Serialize};
+use tokio_postgres::Transaction;
+use utoipa::ToSchema;
+
+#[derive(Deserialize, Serialize, Debug, Clone, ToSchema)]
+pub struct Expiration {
+    pub deletion_timestamp: Option<DateTime>,
+    pub delete_record: bool,
```

Comment:
does it make sense to delete the dataset record without deleting the data? Maybe make it an enum DeleteRecordAndData, DeleteRecord?
or throw an an error if delete_record ist true but delete_data is false?

Reply:
I don't think it does, but that's how GeoEngine works, so I originally kept the option for it. I changed it now to the suggested enum (7b4fa9c2b0233db64d3505387f34555d02c2b069 c56db3069fa8e7e6bc42f6a2626134a21a82ce95 3fc317a8357d323dec83b88a67c4419ca9dc507f). An odd part now is delete_dataset from the DatasetStore, I added a new comment there, feel free to comment or resolve if this is not a problem: https://github.com/geo-engine/geoengine/pull/970/files#r1672610833